### PR TITLE
Shrestha/integrate remove ng assign

### PR DESCRIPTION
**Pull request by shresthamalik**
- RewritePass has an additional phase after EnterInCatalog to RemoveNGraphAssigns. 
- EncapsulateOp directly updates the Variable with the computed value
- Compute Op output tensors are retrieved with the help of Catalog

### CODE REVIEW

File: ngraph_bridge/enable_variable_ops/ngraph_enter_in_catalog.h [hunk 1]
```diff
@@ -50,11 +50,12 @@ namespace ngraph_bridge {
...
-// Op, we store this NG-Tensor
-// so that it can be directly accessed in compute call of NGraphVariableType.
-// We add mapping of {graphId_encapnodename_OutputIndex : NG-Tensor} to the
-// EncapOutputTensorMap
+// 2. If the input to NGraphAssign Op is from NGraphEncapsulate Op
+// We add mapping of
+// {graphId_encapnodename_OutputIndex : tuple:{Variable_Shared_Name, CopyToTF,
+// IsTFJustLooking}}
+// to the EncapOutputInfoMap
+// We attach "_ngraph_remove" attribute to the NGraphAssign node
```
Comment (shresthamalik) [inline ngraph_bridge/enable_variable_ops/ngraph_enter_in_catalog.h]: ``
```suggestion
// We attach "_ngraph_remove" attribute to this NGraphAssign node
```

File: ngraph_bridge/enable_variable_ops/ngraph_assign_op.cc [hunk 2]
```diff
@@ -124,19 +130,19 @@ class NGraphAssignOp : public OpKernel {
...
-      auto ng_val = NGraphCatalog::GetTensorFromEncapOutputTensorMap(valkey);
-      var->update_ng_tensor(ng_val);
-    } else {
-      NGRAPH_VLOG(4) << "NGraphAssign::Getting from TF : " << valkey;
-      if (var->update_ng_tensor(rhs_tensor)) {
-        number_of_copies++;
-        copy_log_str << " COPY_INP_VAL[0]";
-      }
+
+    // input[1] cannot be from NGraphEncap Op
```
Comment (kanvi-nervana) [inline ngraph_bridge/enable_variable_ops/ngraph_assign_op.cc]: May be give more details here

File: ngraph_bridge/ngraph_encapsulate_op.cc [hunk 3]
```diff
@@ -788,22 +831,50 @@ class NGraphEncapsulateOp : public OpKernel {
 #if defined(NGRAPH_TF_ENABLE_VARIABLES_AND_OPTIMIZERS)
       if (m_number_outputs == -1) {
         NGRAPH_VLOG(4) << "Settig number of outputs for " << def().name();
-        m_number_outputs = output_caches.size();
+        m_number_outputs = ng_outputs.size();
+        NGRAPH_VLOG(4) << "Settig number of inputs for " << def().name();
```
Comment (kanvi-nervana) [inline ngraph_bridge/ngraph_encapsulate_op.cc]:
```suggestion
        NGRAPH_VLOG(4) << "Setting number of inputs for " << def().name();
```

File: ngraph_bridge/ngraph_encapsulate_op.cc [hunk 4]
```diff
@@ -788,22 +831,50 @@ class NGraphEncapsulateOp : public OpKernel {
...
+                  output_key);
+          NGraphVar* var;
+          OP_REQUIRES_OK(ctx, ctx->resource_manager()->Lookup<NGraphVar>(
+                                  ctx->resource_manager()->default_container(),
+                                  ref_var_name, &var));
+
+          if (NGraphCatalog::GetCopyToTFFromEncapOutputInfoMap(output_key)) {
+            if (var->copy_ng_to_tf()) {
+              number_of_copies++;
+              copy_log_str << " COPY_TF ";
```
Comment (kanvi-nervana) [inline ngraph_bridge/ngraph_encapsulate_op.cc]: maybe change
copy_log_str << " COPY_TF ";   ----->   copy_log_str << " COPY_TO_TF ";

**Pull request merged**